wwwapp/models.py

```diff
@@ -295,9 +295,13 @@ class Workshop(models.Model):
     qualification_threshold = models.DecimalField(null=True, blank=True, decimal_places=1, max_digits=5)
     max_points = models.DecimalField(null=True, blank=True, decimal_places=1, max_digits=5)
 
+    def is_editable(self):
+        return (not hasattr(self, 'type') or self.type.year == settings.CURRENT_YEAR) and \
+               datetime.datetime.now().date() < settings.WORKSHOPS_START_DATE
+
     def clean(self):
         super(Workshop, self).clean()
-        if self.type.year != settings.CURRENT_YEAR:
+        if not self.is_editable():
```

Comment:
1. Can I get here an override for superusers? (I don't like when system says I can't change something and there is no override)
2. What if somebody wants to edit an workshop after workshops have stared? Cases I can think of: mark workshops as cancelled, add some links or data for participants.

Reply:
1. Yeah sure, I actually had to modify settings.CURRENT_YEAR while testing this :joy: A warning saying "you are editing past workshops" would be nice though. We will also have to modify the existing save() method in the model as that blocks changes even through the Django admin page, not to mention normal forms.
2. That's actually a good point. I guess it should allow changes until CURRENT_YEAR is changed.

Reply:
1. Actually this is a bigger problem, e.g. you can't edit qualification results as an admin either. We could fix that as well while working on it.